core/trino-main/src/main/java/io/trino/json/ir/IrNamedValueVariable.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.json.ir;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import io.trino.spi.type.Type;
+
+import java.util.Objects;
+import java.util.Optional;
+
+import static java.util.Objects.requireNonNull;
+
+public class IrNamedValueVariable
```

Comment:
what is the benefit of having separate `IrNamedValueVariable` and `IrNamedJsonVariable`

Reply:
Having separate classes lets me do sanity checks on the actual values. Those checks will be even more needed when the path tree is transformed by optimizations.
I'd rather keep them separate now, and schedule it for the future to join them.